test/test_datasets.py

```diff
@@ -1790,5 +1790,78 @@ def test_targets(self):
                 assert item[6] == i // 3
 
 
+class LFWPeopleTestCase(datasets_utils.DatasetTestCase):
+    DATASET_CLASS = datasets.LFWPeople
+    FEATURE_TYPES = (PIL.Image.Image, int)
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        train=(True, False),
+        image_set=('original', 'funneled', 'deepfunneled')
+    )
+    _IMAGES_DIR = {
+        "original": "lfw",
+        "funneled": "lfw_funneled",
+        "deepfunneled": "lfw-deepfunneled"
+    }
+
+    def inject_fake_data(self, tmpdir, config):
+        tmpdir = pathlib.Path(tmpdir) / "lfw-py"
+        os.makedirs(tmpdir, exist_ok=True)
+        split = "Train" if config["train"] else "Test"
+        return dict(
+            num_examples=self._create_images_dir(tmpdir, self._IMAGES_DIR[config["image_set"]], split),
+            split=split
+        )
+
+    def _create_images_dir(self, root, idir, split):
+        idir = os.path.join(root, idir)
+        os.makedirs(idir, exist_ok=True)
+        num_people = 4
+        num_examples = 0
+        flines = [str(num_people)]
+        for i in range(num_people):
+            name = self._create_random_id()
+            no = random.randint(1, 10)
+            flines.append(f"\n{name}\t{no}")
+            datasets_utils.create_image_folder(idir, name, lambda n: f"{name}_{n+1:04d}.jpg", no, 250)
+            num_examples += no
+        with open(pathlib.Path(root) / f"peopleDev{split}.txt", "w") as f:
+            f.writelines(flines)
+
+        return num_examples
+
+    def _create_random_id(self):
+        return datasets_utils.create_random_string(random.randint(5, 7)) + '_' \
+            + datasets_utils.create_random_string(random.randint(4, 7))
```

Comment:
Nit: let's try to avoid backslashes where ever possible

```suggestion
    def _create_random_id(self):
        part1 = datasets_utils.create_random_string(random.randint(5, 7))
        part2 = datasets_utils.create_random_string(random.randint(4, 7))
        return f"{part1}_{part2}"
```